services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/resource_tracker_core.py

```diff
@@ -1,135 +1,91 @@
-                "image": image,
-                "user_id": userid,
-            }
+async def _prometheus_client_custom_query(
+    prometheus_client: PrometheusConnect, promql_cpu_query: str
+) -> list[dict]:
+    _logger.info("Querying prometheus with: %s", promql_cpu_query)
+    data: list[dict] = await asyncio.get_event_loop().run_in_executor(
+        None, prometheus_client.custom_query(promql_cpu_query)
+    )
+    return data
 
 
-async def _evaluate_service_resource_usage(
+async def _scrape_and_upload_container_resource_usage(
     prometheus_client: PrometheusConnect,
-    start_time: datetime.datetime,
-    stop_time: datetime.datetime,
-    user_id: UserID,
-    uuid: str = ".*",
-    image: str = "registry.osparc.io/simcore/services/dynamic/jupyter-smash:3.0.9",
-) -> dict[str, Any]:
-    max_values_per_docker_id: dict[str, Any] = {}
-    time_delta = stop_time - start_time
-    minutes = round(time_delta.total_seconds() / 60)
+    resource_tracker_repo: ResourceTrackerRepository,
+    image_regex: str = "registry.osparc.io/simcore/services/dynamic/jupyter-smash:.*",
+) -> None:
+    # Query CPU seconds
+    promql_cpu_query = f"sum without (cpu) (container_cpu_usage_seconds_total{{image=~'{image_regex}'}})[30m:1m]"
+    containers_cpu_seconds_usage: list = await _prometheus_client_custom_query(
+        prometheus_client, promql_cpu_query
+    )
+    _logger.info(
+        "Recieved %s containers from Prometheus", len(containers_cpu_seconds_usage)
+    )
 
-    for current_datetime in [
-        stop_time - datetime.timedelta(minutes=i) for i in range(minutes)
-    ]:
-        rfc3339_str = current_datetime.isoformat("T")
-        # Query CPU seconds
-        promql_cpu_query = f"sum without (cpu) (container_cpu_usage_seconds_total{{container_label_user_id='{user_id}',image='{image}',container_label_uuid=~'{uuid}'}})"
-        container_cpu_seconds_usage = prometheus_client.custom_query(
-            promql_cpu_query, params={"time": rfc3339_str}
-        )
-        # Query network egress
-        promql_network_query = f"container_network_transmit_bytes_total{{container_label_user_id='{user_id}',image='{image}',container_label_uuid=~'{uuid}'}}"
-        container_network_egress = prometheus_client.custom_query(
-            promql_network_query, params={"time": rfc3339_str}
+    for item in containers_cpu_seconds_usage:
+        # Prepare metric
+        metric: dict[str, Any] = item["metric"]
+        container_label_simcore_service_settings: list[dict[str, Any]] = json.loads(
+            metric["container_label_simcore_service_settings"]
         )
+        cpu_reservation: int | None = None
+        ram_reservation: int | None = None
+        for setting in container_label_simcore_service_settings:
+            if setting.get("type") == "Resources":
+                cpu_reservation = int(setting["value"]["Reservations"]["NanoCPUs"])
+                ram_reservation = int(setting["value"]["Reservations"]["MemoryBytes"])
```

Comment:
so this is a Docker Service information, normally not available on Docker containers (as I said containers only have a limit, no reservation. so maybe let's check together to prevent confusion here)